src/Navigation/Notifications/Entity/Notify.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace Friendica\Navigation\Notifications\Entity;
+
+use DateTime;
+use Friendica\BaseEntity;
+use Friendica\Content\Text\BBCode;
+use Friendica\Core\Renderer;
+use Psr\Http\Message\UriInterface;
+
+/**
+ * @property-read $type
+ * @property-read $name
+ * @property-read $url
+ * @property-read $photo
+ * @property-read $date
+ * @property-read $msg
+ * @property-read $uid
+ * @property-read $link
+ * @property-read $itemId
+ * @property-read $parent
+ * @property-read $seen
+ * @property-read $verb
+ * @property-read $otype
+ * @property-read $name_cache
+ * @property-read $msg_cache
+ * @property-read $uriId
+ * @property-read $parentUriId
+ * @property-read $id
+ */
+class Notify extends BaseEntity
+{
+	/** @var int */
+	protected $type;
+	/** @var string */
+	protected $name;
+	/** @var UriInterface */
+	protected $url;
+	/** @var UriInterface */
+	protected $photo;
+	/** @var DateTime */
+	protected $date;
+	/** @var string */
+	protected $msg;
+	/** @var int */
+	protected $uid;
+	/** @var UriInterface */
+	protected $link;
+	/** @var int */
+	protected $itemId;
+	/** @var int */
+	protected $parent;
+	/** @var bool */
+	protected $seen;
+	/** @var string */
+	protected $verb;
+	/** @var string */
+	protected $otype;
+	/** @var string */
+	protected $name_cache;
+	/** @var string */
+	protected $msg_cache;
+	/** @var int */
+	protected $uriId;
+	/** @var int */
+	protected $parentUriId;
+	/** @var int */
+	protected $id;
+
+	public function __construct(int $type, string $name, UriInterface $url, UriInterface $photo, DateTime $date, int $uid, UriInterface $link, bool $seen, string $verb, string $otype, string $name_cache, string $msg = null, string $msg_cache = null, int $itemId = null, int $uriId = null, int $parent = null, int $parentUriId = null, int $id = null)
+	{
+		$this->type        = $type;
+		$this->name        = $name;
+		$this->url         = $url;
+		$this->photo       = $photo;
+		$this->date        = $date;
+		$this->msg         = $msg;
+		$this->uid         = $uid;
+		$this->link        = $link;
+		$this->itemId      = $itemId;
+		$this->parent      = $parent;
+		$this->seen        = $seen;
+		$this->verb        = $verb;
+		$this->otype       = $otype;
+		$this->name_cache  = $name_cache;
+		$this->msg_cache   = $msg_cache;
+		$this->uriId       = $uriId;
+		$this->parentUriId = $parentUriId;
+		$this->id          = $id;
+	}
+
+	public function setSeen()
+	{
+		$this->seen = true;
+	}
+
+	public function updateMsgFromPreamble($epreamble)
+	{
+		$this->msg       = Renderer::replaceMacros($epreamble, ['$itemlink' => $this->link->__toString()]);
```

Comment:
We should mark this for later, because it breaks the pattern here :-) .. I don't know how many dependencies `Renderer` has, but it injects them with that call into the "small" data-classes

Reply:
This is fair enough, the smallest way to handle this would be to have a `Renderer` object we can pass to the method, right?

Reply:
Yes, or a more aadvanced would be to ask ourself why we use render inside "small" data entities, maybe it's part of the caller to render the msg (just an idea)

Reply:
I fully agree.

Reply:
So I looked into this, and having the caller render the message would pretty much pare down the method to just a couple of arbitrary property changes, which I'm not too happy about it.

But it's currently part of `include/enotify` which is slated to be refactored away, just wait until I understand why it is so complicated.